docs/resources/client/i18n.md

```diff
@@ -120,6 +120,31 @@ Some places in Minecraft offer you helper methods to get a translation keys. For
 The only purpose of translation keys is for localization. Do not use them for game logic, that's what [registry names][regname] are for.
 :::
 
+### Component Translations
+
+NeoForge extends the vanilla language file format to allow using arrays of `Component`s in translations.
+Additionally, NeoForge registers the inserting component (type `neoforge:inserting`), which inserts
+an argument from the surrounding translation.
```

Comment:
This is not explained at all and I have no idea what it does from such a brief explanation.